### DISCUSSION TITLE
Monthly ESMValTool community meeting October 2022

### DISCUSSION THREAD

**Discussion by bouweandela** [ESMValGroup/Community#49] [Meetings]
Dear @ESMValGroup/esmvaltool-developmentteam,

The next monthly ESMValTool community meeting will be in October. The result of the poll is that the meeting will take place on

**Tuesday, October 11 at 2 PM CEST**

The duration will be one hour. [Join the zoom meeting](https://us02web.zoom.us/j/86356385817).

Calendar invitations to the meeting are sent to the mailing list, join it [here](https://docs.esmvaltool.org/en/latest/introduction.html#user-mailing-list).

Format and agenda

- Anyone with an interest in ESMValTool is welcome to join and ask questions
- We start the meeting by doing a round where everyone gets one minute to introduce themselves and tell what they will be working on in the next month or what they would like help with.
- Using the input from the introduction round, we choose some topics to talk about in more detail.
- If you already know that you would like to bring up a particular topic in the meeting, you can mention the [GitHub issue](https://github.com/ESMValGroup/ESMValTool/issues) on the topic in a comment below. That will make it easy to find for the other people in the meeting and we can already start discussing there before the meeting. Just [create a new GitHub issue](https://github.com/ESMValGroup/ESMValTool/issues/new) about your topic if there is no open issue yet.
- GitHub issues are our preferred way to communicate outside the meeting and keep track of work, so any feature requests/action items/etc discussed in the meeting should ideally result in a GitHub issue.

Notes from the previous meeting are available in https://github.com/ESMValGroup/Community/discussions/46#discussioncomment-3595863

**Comment by bouweandela** [comment]
@malininae Apologies for the early timeslot, next time I will try to pick a later timeslot.

**Reply by malininae** [reply]
> @malininae Apologies for the early timeslot, next time I will try to pick a later timeslot.

@bouweandela no worries! I'll gather a bunch of stuff for November one :-)

**Comment by bouweandela** [comment]
Update round summary

@rswamina Starting on a project for hot model evaluation
@ehogan Preparing for the upcoming workshop
@sloosvel and @pepcos Metadata fixes in CORDEX, helping @valeriupredoi 
@glpotter and @valeriupredoi Finished the MERRA2 CMORizer https://github.com/ESMValGroup/ESMValTool/pull/2774, starting on a new CMORizer for CFSR, also interested in making ERA5 data available
@valeriupredoi Working on the v2.7 release: release candidate of ESMValCore v2.7 available
@AnushaNTNU Comparing CMIP6 for future scenarios
@bouweandela Working on improved support for specifying ancillary datasets in the recipe, dataset versioning and extending the Python API with functionality for finding and loading data (design work in progress in https://github.com/ESMValGroup/ESMValCore/pull/1609) and improving the computational performance by making better use of dask (draft of a new parallel running mode here https://github.com/ESMValGroup/ESMValCore/pull/1714).

**Comment by bouweandela** [comment]
Upcoming v2.7 release update by @valeriupredoi with the help of @sloosvel and @remi-kazeroni 
- ESMValCore v2.7 release candidate package available
- @valeriupredoi Is running recipes to test ESMValCore release candidate, so far very few issues encountered
- Please suggest highlights ESMValGroup/ESMValTool#2833
- Expect ESMValCore release can be expected by Thursday, October 13
- Mostly using Levante because more observational data is available there and it seems less busy
- Comparison of recipe results to previous version results will be done when testing for ESMValTool release

**Comment by bouweandela** [comment]
Update by @glpotter:

- His time at NASA will end in a few weeks time
- Finishing up reanalyses of CREATE-IP so they run up to 2020
- Would like to work on CFSR, needs conversion from .grib to .nc. This should be straightforward using [iris-grib](https://iris-grib.readthedocs.io/en/stable/) with our [CMORizer framework](https://docs.esmvaltool.org/en/latest/input.html#using-a-cmorizer-script). @valeriupredoi offered to help out.

**Comment by bouweandela** [comment]
About the workshop:

- @glpotter asked if the meetings could be recorded for people from the US (LLNL) who are interested in using the tool. @ehogan will find out
- Agenda is available here: https://docs.google.com/document/d/1iIBZnH0lV04XIqJC1XAviv5Fm5hO3MJkF44kR-nUMMw/edit
- See https://github.com/ESMValGroup/Community/discussions/45 for all the information on the workshop

**Reply by malininae** [reply]
@bouweandela @ehogan , if there would be a possibility to record some of the pre-lunch discussions, that would be amazing!

**Comment by bouweandela** [comment]
@anushaNTNU is starting to use the tool and could use some help

- Tried a few recipes, but is interested in using them with future scenarios instead of historical. When changing the recipes, they don't work: @bouweandela said that usually this can be fixed by making a small change to the code of the diagnostic. It's best to install from source by following the instructions [here](https://docs.esmvaltool.org/en/latest/quickstart/installation.html#install-from-source). You can then edit the diagnostic script to make it work for your use case. When a Python diagnostic crashes, it will show a [traceback](https://realpython.com/python-traceback/#how-do-you-read-a-python-traceback) with the filename and the line of code on which the error occurred.
- In the tutorial, there is also a chapter on doing an [installation with the source code](https://esmvalgroup.github.io/ESMValTool_Tutorial/07-development-setup/index.html) and a [chapter that introduces how to write your own recipe and diagnostic](https://esmvalgroup.github.io/ESMValTool_Tutorial/06-preprocessor/index.html), which may help understand how recipes and diagnostics work. A list of (relatively) simple example recipes can be found in the [contributing a recipe](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#creating-a-recipe-and-diagnostic-script-s) documentation, but most of that chapter is targeted at people who actually want to contribute a recipe and/or diagnostic, so this may be a bit too involved if you're just starting to learn the tool.
- @rswamina Offered to help with specific problems :tada:
